src/plonk_recursion.rs

```diff
@@ -47,9 +48,100 @@ impl RecursionPublicInputs {
             self.o_plonk_t.as_slice(),
             self.halo_u_l.as_slice(),
             self.halo_u_r.as_slice(),
+            self.old_proofs.as_slice(),
         ]
         .concat()
     }
+
+    pub fn populate_witness<C: Curve>(
```

Comment:
Hmm my expectation was that callers would populate the `ProofTarget` and everything else would be generated by the circuit. The code near the end of `recursive_verification_circuit` should copy claimed openings from `ProofTarget`'s wire to the public input wires, and the challenge elements should be generated by `Rescue*` gates. So I don't think we really need these steps, but let me know if the generators aren't working properly and I can debug.